lib/plugin.js

```diff
@@ -82,55 +106,59 @@ Plugin.prototype.exec = function (server, cb) {
       clearTimeout(timer)
     }
 
-    cb(err)
+    callback(execErr)
   }
 
   if (this.timeout > 0) {
     debug('setting up timeout', name, this.timeout)
     timer = setTimeout(function () {
       debug('timed out', name)
       timer = null
-      const err = new AVV_ERR_READY_TIMEOUT(name)
-      err.fn = func
-      done(err)
+      const readyTimeoutErr = new AVV_ERR_READY_TIMEOUT(name)
```

Comment:
Wouldnt it make more sense to create and use a AVV_ERR_PLUGIN_EXEC_TIMEOUT?

Reply:
Why?

Reply:
Imho there are two timeouts. One is that the plugin loaded too long. For this I throw AVV_ERR_PLUGIN_EXEC_TIMEOUT. The other timeout is when the overall loading takes simply too long, e.g. all plugins take together more time than the avvio is allowed to.

I think it should be differenciated. But I am willing to revert that change.

Reply:
@metcoder95 

We should discuss here if the new Error makes more sense.